### PR TITLE
cleanup build.gradle

### PR DESCRIPTION
**Pull request by dulmandakh**
cleanup build.gradle to use gradle settings from a project.
- remove gradle plugin specification, because we don't need it in the library as we don't build AAR. This caused Android Studio to remove **react-native-linear-gradient** from a project because it uses different gradle plugin version.
- use latest version (+) of react-native as compileOnly dependency

@sibelius, please merge and release a new version 😄

### CODE REVIEW

**Comment by sibelius** [comment]
it should be `compileOnly`

check react-native-camera build.gradle

https://github.com/react-native-community/react-native-camera/blob/master/android/build.gradle#L56

**Comment by dulmandakh** [comment]
@sibelius please merge and release a new version

**Comment by sibelius** [comment]
compile will cause problems

It should be compileOnly

**Comment by dulmandakh** [comment]
I tested it for myself and it just works. compileOnly is only available if you use gradle plugin 3.x, which obsoletes compile and provided dependencies.

why push people to use gradle plugin 3.x, while RN uses 2.x?

**Comment by sibelius** [comment]
Gradle 3 will be required in the end of this year

We should start migrating to i

It improves build time

Compile does not work well with gradle 3

**Comment by dulmandakh** [comment]
it compiles with gralde plugin 3.x, but with warning.

I think we can land gradle plugin 3.x in RN by the end of the year. RN needs to fix deprecation warning and aapt2 before gradle plugin 3.x.

**Comment by sibelius** [comment]
there is gradle 4 already https://docs.gradle.org/4.0/release-notes.html

**Comment by dulmandakh** [comment]
There is gradle and gradle plugin. RN uses gradle 3.x, but gradle plugin 2.x https://github.com/facebook/react-native/commit/699e5eebe807d1ced660d2d2f39b5679d26925da

Also, there is a PR for gradle 4.x https://github.com/facebook/react-native/pull/19758.

Need some more work to make gradle plugin 3.x work.

**Comment by dulmandakh** [comment]
@sibelius bumped build tools to 26.0.3, because RN landed it https://github.com/facebook/react-native/commit/b569154ae633d7f3c6b4f148b108f05aef742bf9

File: android/build.gradle [hunk 1]
```diff
@@ -1,39 +1,22 @@
...
 android {
-    compileSdkVersion _compileSdkVersion
-    buildToolsVersion _buildToolsVersion
+    compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
+    buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
 
     defaultConfig {
-        minSdkVersion _minSdkVersion
-        targetSdkVersion _targetSdkVersion
+        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.targetSdkVersion : DEFAULT_MIN_SDK_VERSION
```
[the remark this answers is not in the document]
Comment (dulmandakh) [reply android/build.gradle]: Thank you. Fixed

**Comment by dulmandakh** [comment]
@sibelius please merge and release a new version

**Comment by dulmandakh** [comment]
landed project wide properties to RN core 
https://github.com/facebook/react-native/commit/0a3055d98a36e49746144e883edc7e20afec4fcb

**Comment by dulmandakh** [comment]
Please merge and release new version.

CC @brentvatne @satya164 @grabbou

**Comment by dulmandakh** [comment]
Or please add me as a contributor, so I can help with Android issues.

**Comment by dulmandakh** [comment]
Please review and merge this. It's ruining my developer experience because I have to edit the gradle file every time I install or update npm package.

@dabit3 @dustinsavery @sibelius @GantMan @ide @brentvatne @ivanzotov @knowbody @Kureev @turnrye @oblador @magicismight

**Comment by satya164** [comment]
LGTM. Makes sense that the lib should support same version of gradle plugin that React Native uses.

**Comment by dulmandakh** [comment]
@satya164 we don't need to specify gradle plugin in library, so omitting it. Gradle plugin mismatch between library and project is causing troubles. Please merge

**Comment by satya164** [comment]
I'll let the project owners merge. But the change looks good to me.

**Comment by dulmandakh** [comment]
We're working to land latest version of gradle and Android SDK. Once landed I'll create a PR to add support for it https://github.com/facebook/react-native/pull/20026

**Comment by ryuoryuo** [comment]
Any news?

**Pull request merged**